### PR TITLE
Add back moe backend

### PR DESCRIPTION
**Pull request by Berstanio**
The generator creates a nearly fully functional MOE backend. The only thing that the user needs to configure after the generation is 1. configuring signing and 2. include the xcframeworks for linking in the "xcode/native/ios/" folder.

It would be also great if it could run two gradle tasks directly after the generation, but I don't know whether this is possible.

### CODE REVIEW

File: src/main/kotlin/gdx/liftoff/data/platforms/iOSMOE.kt [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    init {
+        dependencies.add("project(':${Core.ID}')")
+        addDependency("io.github.berstanio:gdx-backend-moe:\$gdxVersion")
+        addSpecialDependency("natives \"com.badlogicgames.gdx:gdx-platform:\$gdxVersion:natives-ios\"")
+    }
+
+    override fun getContent() = """apply plugin: 'moe'
+
+// Exclude all files from Gradle's test runner
+test { exclude '**' }
```
Comment (tommyettinger) [inline src/main/kotlin/gdx/liftoff/data/platforms/iOSMOE.kt:144]: I'm a little uncertain about this, but if I am reading this right, this only disables tests in the MOE module? (And they can still be run in other ways, I guess).

File: src/main/resources/i18n/nls.properties [hunk 2]
```diff
@@ -3,6 +3,7 @@ core=Core
 lwjgl2=Desktop (Legacy)
 lwjgl3=Desktop
 ios=iOS
+ios-moe=iOS Multi-Os Engine
```
Comment (tommyettinger) [inline src/main/resources/i18n/nls.properties:6]: Pretty sure this should be OS (Operating System), but it isn't a big deal, I'll fix it after I merge.

**Review by tommyettinger — APPROVED** [review]
All around, this looks pretty good. I'm not sure if the large images are the right way to go with the App Store right now, but if those same images already exist in RoboVM-iOS assets, then we can reuse them. What are the two Gradle tasks? I'll merge anyway, but I think I can make this run a few at the end of creating a project (as long as they aren't insanely slow).

File: src/main/kotlin/gdx/liftoff/data/platforms/iOSMOE.kt (hunk 1 above)
Comment (Berstanio) [reply src/main/kotlin/gdx/liftoff/data/platforms/iOSMOE.kt:144]: Yes, exactly!

File: src/main/resources/i18n/nls.properties (hunk 2 above)
Comment (Berstanio) [reply src/main/resources/i18n/nls.properties:6]: Ahh yes, my bad! Thanks!

**Comment by Berstanio** [comment]
> I'm not sure if the large images are the right way to go with the App Store right now, but if those same images already exist in RoboVM-iOS assets, then we can reuse them.

Yes, they are. And also, it is probably something the user will change when publishing to App Store.



> What are the two Gradle tasks? I'll merge anyway, but I think I can make this run a few at the end of creating a project (as long as they aren't insanely slow).

// Just for the history
Ahh sure, sryy forgot that!
The 1. one is "moeUpdateXcodeSettings". This will update the xcode project files according to the currently used moe release, because they can change every release. With just running the task on generation would let liftoff keep a old project file and just update it on generation.
The 2. one would be the "copyNatives" task, so that the frameworks are from the beginning there to use/include in xcode.

They both should be very fast, so it shouldn't make any difference.

**Comment by tommyettinger** [comment]
It looks like `moeUpdateXcodeSettings` isn't an available task for me on Windows, so I'm not sure if it's possible to auto-run it on all OSes. It also may just be missing for some other reason. I believe copyNatives works upon project import, but I'm not sure.

**Comment by Berstanio** [comment]
> It looks like `moeUpdateXcodeSettings` isn't an available task for me on Windows, so I'm not sure if it's possible to auto-run it on all OSes. It also may just be missing for some other reason. I believe copyNatives works upon project import, but I'm not sure.

I confirmed that this is a valid task also on windows. What error message do you get when running it on your machine?

**Comment by Berstanio** [comment]
Probably not, as far as I can tell this code seems strange to me:
https://github.com/tommyettinger/gdx-liftoff/blob/5eec595b8d6f2e81e008d1092755d1fbd14a960e/src/main/kotlin/gdx/liftoff/views/AdvancedProjectData.kt#L78-L80
I'm not that familiar with kotlin, but won't this create a new list on every get execution? Therefor the list of tasks to execute wont be saved?
It seems this way, at least this is what I get with debugging.

You could check whether "copyNatives" was also run, So if dirs are under "ios-moe/xcode/natives/ios/".
Or you could check whether project.pbxproj file changes when you execute `moeUpdateXcodeSettings ` a second time.

**Comment by tommyettinger** [comment]
It was strange to me too; I had to change that to use mutable lists, but I am not very knowledgeable about how Kotlin's details work, and it probably is creating a new MutableList each time the property is accessed. I'll take a look at your PR now.

**Pull request merged**